frontend/src/api/types.ts

```diff
@@ -1,7 +1,17 @@
 // The module might be removed if we start using open api client generator
 
-export interface ExperimentalMeasurementDto {
+export interface MeasurementDto {
   compoundId: string,
   concentration: number,
   signal: number
 }
+
+export type CompoundsMap = { [compoundId: string]: MeasurementDto[] };
+
+export function groupCompounds(mes: MeasurementDto[]): CompoundsMap {
+  return mes.reduce((rv: CompoundsMap, x) => {
+    // eslint-disable-next-line no-param-reassign
+    (rv[x.compoundId] = rv[x.compoundId] || []).push(x);
```

Comment:
```suggestion
    const rvReald = rv[x.compoundId] || []
    rvReald.push(x);
    return rvReald;
```
Also what mean `rv`? May you please rename it more understandable